application/helpers/update/updatedb_helper.php

```diff
@@ -2542,6 +2542,23 @@ function db_upgrade_all($iOldDBVersion, $bSilent = false)
             $oTransaction->commit();
         }
 
+        /**
+         * Sanitize theme option paths
+         */
+        if ($iOldDBVersion < 366) {
+            $oTransaction = $oDB->beginTransaction();
+            
+            $templateConfigurations = TemplateConfiguration::model()->findAll();
```

Comment:
I know model are not supposed to be handled here, but we have not many other ways to do it.

Reply:
SQL ? No ?

` $templateConfigurations = "SELECT *  from {{template_configuration}} WHERE id > 0";`

id > 0 because i have an issue with WHERE 1 on MSSQL (plugin)

Reply:
OK, got the reason … clearly : doing something with model inside update broke some upgrade part …

It can be done at the 1st load ? Adding a "version" in the model ? If version is 2 : it's done  ?

Reply:
I am sorry. I don't understand. Can you provide a bit more context please?

Reply:
If you start with a version without TemplateConfiguration : you broke update …

**Need** another solution …

Reply:
> you broke update

How I breake it?

Reply:
Please : setup a 2.6 and update. It broke at this step.

It's the reason you must not use model in update.

Reply:
And that would be solved by using this?
`$templateConfigurations = "SELECT * from {{template_configuration}} WHERE id > 0";`

There is a lot of logic in the update that is in the model.
Want to avoid code duplication.

Reply:
1st look :, second one is here : https://github.com/LimeSurvey/LimeSurvey/pull/1927#discussion_r653246334

Reply:
> Please : setup a 2.6 and update. It broke at this step.

We have tried. No error happened.
Can I ask you to elaborate more? I am sorry but I can't understand what's wrong with short answers.

Reply:
> We have tried. No error happened.

? Strange and great then :)